app/controllers/schedulers_controller.rb

```diff
@@ -87,9 +103,29 @@ def visual_run
   action_auth_level :update, :instructor
   def update
     @scheduler = Scheduler.find_by(id: params[:id])
-    if @scheduler&.update(scheduler_params)
-      flash[:success] = "Scheduler updated."
-      redirect_to(course_schedulers_path(@course))
+    # Check if the action file exists and is readable
+    action_path = Rails.root.join(scheduler_params[:action]).to_path
```

Comment:
this kind of controller code seems fine for the most part, although it is a bit funky to update the scheduler to do the visual run and then revert afterwards if it didn't work, but I guess there's not a super clean way of doing this?